### ISSUE TITLE
Docker healthcheck fails with version 1.28 because of 'curl: command not found'

### ISSUE THREAD

**Issue by bruegth** [envoyproxy/envoy#31614]
Looks like `curl` command is no longer included in docker image and the HEALTHCHECK will fail:
`"Output": "curl: command not found\n" `
https://github.com/envoyproxy/envoy/blob/4ed411184d999c3684f7277536c58f7a4058840f/examples/shared/envoy/Dockerfile#L26

This is my workaround (instead of `/stats` I'll prefer `/server_info`):
```
#!/bin/bash
exec 3<>/dev/tcp/localhost/${ENVOY_ADMIN_PORT}

echo -e "GET /server_info HTTP/1.1\nhost: localhost:${ENVOY_ADMIN_PORT}\n" >&3

timeout --preserve-status 1 cat <&3 | grep -m 1 state | grep -m 1 LIVE
ERROR=$?

exec 3<&-
exec 3>&-

exit $ERROR

```

As I see `curl` command is not only used in that example dockerfile, maybe the impact is bigger!

**Comment by phlax** [comment]
curl is installed a few lines previously

https://github.com/envoyproxy/envoy/blob/4ed411184d999c3684f7277536c58f7a4058840f/examples/shared/envoy/Dockerfile#L14

> As I see curl command is not only used in that example dockerfile, maybe the impact is bigger!

not at all - the example has an example of a healthcheck its not part of the distribution

the reason for that is that the healthcheck requires admin

i think this can be closed

**Comment by bruegth** [comment]
Closed in favor of #31619
